microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+                            sudo=True
+                        )
+                        log.debug(
+                            "Commented out GRUB_FORCE_PARTUUID line in "
+                            f"{_40_force_partuuid_file}.")
+                    else:
+                        log.debug("No force part uuid to be replaced, continue test...")
+
+                node.execute(
+                    "update-grub2", sudo=True
+                )
+
+                _entry1 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    'resume=',
+                    sudo=True
+                )
+                assert_that(_entry1).described_as(
+                    f"{_entry1} - Missing config update"
+                    " in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                _entry2 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_HIDDEN_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry2).described_as(
+                    f"{_entry2} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                _entry3 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry3).described_as(
+                    f"{_entry3} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                log.info(
+                    "Successfully updated 50-cloudimg-settings.cfg file"
+                    " with all three entries.\n"
+                    "Setup Hibernate Kernel Completed.")
+
+        #################################
+        #      SETUP HIBERNATE DONE     #
+        #################################
+
+        # Reboot VM to apply swap setup changes
+        log.info("Rebooting VM")
+        start_stop = node.features[StartStop]
+        start_stop.restart()
+        # Check VM status, confirm it's running
+        if not _is_vm_running(node, log, self.REBOOT_TIME_OUT):
+            raise LisaException("Can not identify VM status before hibernate.")
+
+        cmd_res = node.execute(
+            "hwclock --set --date='2033-07-01 12:00:00'",
+            shell=True,
+            sudo=True
+        )
+        before_time_stamp = _get_year(node)
+        log.debug(f"Got before time stamp: {before_time_stamp}")
+
+        node.execute_async(
+            'echo disk > /sys/power/state',
+            shell=True,
+            sudo=True
+        )
+
+        log.info(
+            "Sent hibernate command to the VM and continue checking its status"
+            " in every 15 seconds until 10 minutes timeout.")
```

Comment:
Do you know why it needs to do like this way? Keep sending command for 10 minutes?

Reply:
no the command will be sent only once, we just check vm status periodically and timeout in 10 mins